### PR TITLE
WHO: the_war_doctor.txt and support (TriggerPhaseOutAll)

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

File: forge-game/src/main/java/forge/game/trigger/TriggerPhaseOutAll.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    public String getImportantStackObjects(SpellAbility sa) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(Localizer.getInstance().getMessage("lblPhasedOut")).append(": ");
+        sb.append(sa.getTriggeringObject(AbilityKey.Cards));
+        return sb.toString();
+    }
+
+    private CardCollection filterCards(CardCollection cards) {
+        cards.removeIf(card -> !matchesValidParam("ValidCards", card));
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/trigger/TriggerPhaseOutAll.java:40]: matchesValidParam should already work fine with Iterable like CardCollection

File: forge-gui/res/cardsfolder/upcoming/the_war_doctor.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Name:The War Doctor
+ManaCost:2 R W
+Types:Legendary Creature Time Lord Doctor
+PT:3/5
+T:Mode$ PhaseOutAll | ValidCards$ Permanent.phasedOutOther | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ Whenever one or more other permanents phase out and whenever one or more other cards are put into exile from anywhere, put a time counter on CARDNAME.
+T:Mode$ ChangesZoneAll | ValidCards$ Card.Other | Origin$ Any | Destination$ Exile | TriggerZones$ Battlefield | Execute$ TrigPutCounter | Secondary$ True | TriggerDescription$ Whenever one or more other permanents phase out and whenever one or more other cards are put into exile from anywhere, put a time counter on CARDNAME.
+SVar:TrigPutCounter:DB$ PutCounter | CounterType$ TIME
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigDamage | TriggerDescription$ Whenever CARDNAME attacks, it deals damage equal to the number of time counters on it to any target. If a creature dealt damage this way would die this turn, exile it instead.
+SVar:TrigDamage:DB$ DealDamage | ValidTgts$ Any | DamageSource$ TriggeredAttackerLKICopy | NumDmg$ Count$CardCounters.TIME
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_war_doctor.txt:9]: ReplaceDying missing

File: forge-game/src/main/java/forge/game/trigger/TriggerPhaseOutAll.java (hunk 1 above)
Comment (Northmoc) [reply forge-game/src/main/java/forge/game/trigger/TriggerPhaseOutAll.java:40]: hmm, sorry I'm striking out on how to better set up this filter

Comment (tool4ever) [reply forge-game/src/main/java/forge/game/trigger/TriggerPhaseOutAll.java:40]: check `TriggerTokenCreatedOnce` for cleaner implementation

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/trigger/TriggerPhaseOutAll.java:40]: 😬 I see it yikes

**Review by tool4ever — APPROVED** [review]

**Pull request merged**